File: CS-Examples/03_Cells/TraverseCellsValue.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.IO;
using Spire.Xls;
using Spire.Xls.Charts;
using System.Text;
using System.Collections.Generic;

namespace TraverseCellsValue
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Load an existing workbook from a file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\CellValues.xlsx");

            // Get the first worksheet in the workbook
            Worksheet worksheet = workbook.Worksheets[0];

            // Get the collection of cell ranges in the worksheet
            CellRange[] cellRangeCollection = worksheet.Cells;

            // Create a StringBuilder to save the content
            StringBuilder content = new StringBuilder();
            content.AppendLine("Values of the first sheet:");

            // Traverse through the cells and retrieve their values
            foreach (CellRange cellRange in cellRangeCollection)
            {
                // Set the string format for displaying the cell address and value
                string result = string.Format("Cell: " + cellRange.RangeAddress + "   Value: " + cellRange.Value);

                // Add the result string to the StringBuilder
                content.AppendLine(result);
            }

            // Specify the output file name as a txt file
            string outputFile = "Output.txt";

            // Save the content to a txt file
            File.WriteAllText(outputFile, content.ToString());

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launching the output file.
            Viewer(outputFile);
		}
		private void Viewer( string fileName )
		{
			try
			{
				System.Diagnostics.Process.Start(fileName);
			}
			catch{}
		}
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: CS-Examples/03_Cells/TraverseCellsValue.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports System.IO
Imports Spire.Xls
Imports Spire.Xls.Charts
Imports System.Text

Namespace TraverseCellsValue
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Creates a new instance of the Workbook class.
            Dim workbook As New Workbook()

            'Loads an Excel file from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\CellValues.xlsx")

            'Retrieves the first worksheet from the workbook.
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            'Retrieves the collection of all cells in the worksheet.
            Dim cellRangeCollection() As CellRange = worksheet.Cells

            'Creates a new instance of the StringBuilder class for storing the output.
            Dim content As New StringBuilder()
            'Appends a line to the StringBuilder object.
            content.AppendLine("Values of the first sheet:")

            'Iterates through each cell range in the collection.
            For Each cellRange As CellRange In cellRangeCollection
                'Formats the cell address and value into a string.
                Dim result As String = String.Format("Cell: " & cellRange.RangeAddress & "   Value: " & cellRange.Value)

                'Appends the formatted string to the StringBuilder object.
                content.AppendLine(result)
            Next cellRange

            'Specifies the name of the output file.
            Dim outputFile As String = "Output.txt"

            'Writes the contents of the StringBuilder object to the specified file.
            File.WriteAllText(outputFile, content.ToString())
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launching the output file.
            Viewer(outputFile)
		End Sub
		Private Sub Viewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
